media/server/gstplayer/source/tasks/generic/UpdatePlaybackGroup.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 Sky UK
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tasks/generic/UpdatePlaybackGroup.h"
+#include "RialtoServerLogging.h"
+
+namespace firebolt::rialto::server
+{
+UpdatePlaybackGroup::UpdatePlaybackGroup(GenericPlayerContext &context, std::shared_ptr<IGstWrapper> gstWrapper,
+                                         std::shared_ptr<IGlibWrapper> glibWrapper, GstElement *typefind,
+                                         const GstCaps *caps)
+    : m_context{context}, m_gstWrapper{gstWrapper}, m_glibWrapper{glibWrapper}, m_typefind{typefind}, m_caps{caps}
+{
+    RIALTO_SERVER_LOG_DEBUG("Constructing UpdatePlaybackGroup");
+}
+
+UpdatePlaybackGroup::~UpdatePlaybackGroup()
+{
+    RIALTO_SERVER_LOG_DEBUG("UpdatePlaybackGroup finished");
+}
+
+void UpdatePlaybackGroup::execute() const
+{
+    RIALTO_SERVER_LOG_DEBUG("Executing UpdatePlaybackGroup");
+    if (nullptr == m_caps)
+    {
+        RIALTO_SERVER_LOG_DEBUG("Typefind SRC Pad Caps NULL");
+        return;
+    }
+    gchar *typefindCaps = m_gstWrapper->gstCapsToString(m_caps);
+    if (typefindCaps)
+    {
+        RIALTO_SERVER_LOG_DEBUG("Typefind SRC Pad Strm Parsed Caps %s", typefindCaps);
+        if (m_glibWrapper->gStrrstr(typefindCaps, "audio/"))
+        {
+            RIALTO_SERVER_LOG_DEBUG("Typefind Audio Caps %s", typefindCaps);
```

Comment:
Is printing typefindCaps needed here? It was printed one log before.

Reply:
I will remove this print in next commit